README.md
<h1 align="center">
    React mapy.cz
</h1>

<p align="center">
  <a href="https://lundegaard.eu">
    <img alt="by Lundegaard" src="./by-lundegaard.png" width="120" />
  </a>
</p>

<h3 align="center">
Amazing React wrapper of mapy.cz API
</h3>

<p align="center">
A library of components for rendering maps. Goal of this library is to provide means to render flexible maps with multiple features allowing users to visualize exactly what is needed.
</p>

<p align="center">
  <a href="https://github.com/lundegaard/mapy-cz-react">
    <img src="https://flat.badgen.net/badge/-/github?icon=github&label" alt="Github" />
  </a>

   <img src="https://flat.badgen.net/badge/license/MIT/blue" alt="MIT License" />
</p>

## Getting started

Installation `mapy-cz-react` lib as your dependency:

```bash
yarn add mapy-cz-react
```

Or

```bash
npm install mapy-cz-react
```

And look at our [Docs](https://lundegaard.github.io/mapy-cz-react/), right to [Getting started](https://lundegaard.github.io/mapy-cz-react/docs/getting-started), or check our [examples](https://lundegaard.github.io/mapy-cz-react/docs/Examples/01-static-map-with-one-marker) for most common use cases.

## Contribution

We are open to ideas and suggestions! Feel free to make PR!

See
[contribution guide](https://github.com/lundegaard/mapy-cz-react/blob/main/CONTRIBUTING.md)
for guidelines. <br /> <br />

## See our related projects

- [lundium](https://github.com/lundegaard/lundium) - Beautiful React component library.
- [@redux-tools](https://github.com/lundegaard/redux-tools) - Modular
  Redux is possible!
- [react-union](https://github.com/lundegaard/react-union) -
  Intergrate React apps into various CMSs seamlessly.
- [validarium](https://github.com/lundegaard/validarium) - Validations
  done right.

## Licence

This library is published under MIT licence, but it uses Mapy.cz API.<br>
By using this library, you also agree with Mapy.cz [Terms and conditions](http://api.mapy.cz/#pact)

© 2022 Lundegaard a.s.

## TODOs:

- Ensure minimal reliance on `window` object


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### 0.0.4 (2023-06-13)


### Features

* add card element to marker ([#27](https://github.com/MaceM8/react-mapy-cz/issues/27)) ([f1b728a](https://github.com/MaceM8/react-mapy-cz/commit/f1b728afe366e571c43017dd225df29000d1737a))
* add lerna config for publishing ([62517e7](https://github.com/MaceM8/react-mapy-cz/commit/62517e72840f65cd974add75e67b334f83902cec))
* add pointer component ([#23](https://github.com/MaceM8/react-mapy-cz/issues/23)) ([9ed228a](https://github.com/MaceM8/react-mapy-cz/commit/9ed228a7be162a88e2a3937a524bc3df6e6c3d57))
* add route component ([#26](https://github.com/MaceM8/react-mapy-cz/issues/26)) ([a955623](https://github.com/MaceM8/react-mapy-cz/commit/a95562388670bc4cea8c177d19fac17db8d7487d))
* api refactor, added checks ([96b34fe](https://github.com/MaceM8/react-mapy-cz/commit/96b34fe384b03e4b134f1660e647fc2a184f00e1))
* improve loading script, minor refactor ([16778c5](https://github.com/MaceM8/react-mapy-cz/commit/16778c51581b1d29b29e5429c43e6497ff71ce44))
* refactor route components, minor improvements ([e6aca40](https://github.com/MaceM8/react-mapy-cz/commit/e6aca40c6c72c19dc58abac20399ec281ee593bc))


### Bug Fixes

* types, examples ([71f2c7d](https://github.com/MaceM8/react-mapy-cz/commit/71f2c7dc0f549f7d4cabe567edb247a3e8940c0f))

### [0.0.4](https://github.com/MaceM8/react-mapy-cz/compare/v0.0.3...v0.0.4) (2022-04-11)

### 0.0.2 (2022-04-03)

### Features

- add card element to marker ([#27](https://github.com/MaceM8/mapy-cz-react/issues/27)) ([f1b728a](https://github.com/MaceM8/mapy-cz-react/commit/f1b728afe366e571c43017dd225df29000d1737a))
- add pointer component ([#23](https://github.com/MaceM8/mapy-cz-react/issues/23)) ([9ed228a](https://github.com/MaceM8/mapy-cz-react/commit/9ed228a7be162a88e2a3937a524bc3df6e6c3d57))
- add route component ([#26](https://github.com/MaceM8/mapy-cz-react/issues/26)) ([a955623](https://github.com/MaceM8/mapy-cz-react/commit/a95562388670bc4cea8c177d19fac17db8d7487d))
- api refactor, added checks ([96b34fe](https://github.com/MaceM8/mapy-cz-react/commit/96b34fe384b03e4b134f1660e647fc2a184f00e1))
- improve loading script, minor refactor ([16778c5](https://github.com/MaceM8/mapy-cz-react/commit/16778c51581b1d29b29e5429c43e6497ff71ce44))

# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.


website/sidebars.js
/**
 * Creating a sidebar enables you to:
 - create an ordered group of docs
 - render a sidebar for each doc of that group
 - provide next/previous navigation

 The sidebars can be generated from the filesystem, or explicitly defined here.

 Create as many sidebars as you want.
 */

module.exports = {
	// By default, Docusaurus generates a sidebar from the docs folder structure
	// docsSidebar: [{ type: 'autogenerated', dirName: '.' }],

	// But you can create a sidebar manually
	docsSidebar: [
		{
			type: 'doc',
			id: 'getting-started',
			label: 'Getting started',
		},
		{
			type: 'category',
			label: 'Tutorial',
			items: ['Tutorial/quick-start'],
		},
		{
			type: 'category',
			label: 'Examples',
			items: [
				'Examples/01-static-map-with-one-marker',
				'Examples/02-interactive-map',
				'Examples/03-interactive-map-with-layout-switcher',
				'Examples/04-interactive-map-multiple-markers',
				'Examples/05-interactive-map-with-thousands-of-points',
				'Examples/06-interactive-map-with-thousands-of-points-marker',
				'Examples/07-interactive-map-with-clusters-and-signals',
				'Examples/08-interactive-map-with-search',
				'Examples/09-interactive-map-with-pointer',
				'Examples/10-interactive-map-with-route',
				'Examples/11-static-map-with-marker-and-card',
			],
		},
		{
			type: 'category',
			label: 'API',
			items: [
				'API/clusterer',
				'API/keyboard-control',
				'API/layer-control',
				'API/map',
				'API/map-provider',
				'API/marker',
				'API/marker-layer',
				'API/mouse-control',
				'API/multiple-markers',
				'API/pointer',
				'API/route',
				'API/route-layer',
				'API/search-input',
				'API/signals',
				'API/sync-control',
				'API/zoom-control',
			],
		},
	],
};
